drape/font_texture.cpp

```diff
@@ -72,110 +70,194 @@ bool GlyphPacker::CanBePacked(uint32_t glyphsCount, uint32_t width, uint32_t hei
 
 m2::RectF GlyphPacker::MapTextureCoords(const m2::RectU & pixelRect) const
 {
-  float fWidth = static_cast<float>(m_size.x);
-  float fHeight = static_cast<float>(m_size.y);
+  auto const width = static_cast<float>(m_size.x);
+  auto const height = static_cast<float>(m_size.y);
 
-  // Half-pixel offset to eliminate arfefacts on fetching from texture.
+  // Half-pixel offset to eliminate artefacts on fetching from texture.
   float offset = 0.0f;
   if (pixelRect.SizeX() != 0 && pixelRect.SizeY() != 0)
     offset = 0.5f;
 
-  return m2::RectF((pixelRect.minX() + offset) / fWidth,
-                   (pixelRect.minY() + offset) / fHeight,
-                   (pixelRect.maxX() - offset) / fWidth,
-                   (pixelRect.maxY() - offset) / fHeight);
+  return {(pixelRect.minX() + offset) / width,
+          (pixelRect.minY() + offset) / height,
+          (pixelRect.maxX() - offset) / width,
+          (pixelRect.maxY() - offset) / height};
 }
 
 bool GlyphPacker::IsFull() const { return m_isFull; }
 
-GlyphGenerator::GlyphGenerator(ref_ptr<GlyphManager> mng, TCompletionHandler const & completionHandler)
-  : m_mng(mng)
+void GlyphGenerator::GenerateGlyphTask::Run(uint32_t sdfScale)
+{
+  m_generatedGlyphs.reserve(m_glyphs.size());
+  for (auto & data : m_glyphs)
+  {
+    if (m_isCancelled)
+      return;
+
+    auto const g = GlyphManager::GenerateGlyph(data.m_glyph, sdfScale);
+    data.m_glyph.m_image.Destroy();
+    m_generatedGlyphs.emplace_back(GlyphGenerator::GlyphGenerationData{data.m_rect, g});
+  }
+}
+
+void GlyphGenerator::GenerateGlyphTask::DestroyAllGlyphs()
+{
+  for (auto & data : m_glyphs)
+    data.m_glyph.m_image.Destroy();
+
+  for (auto & data : m_generatedGlyphs)
+    data.m_glyph.m_image.Destroy();
```

Comment:
optional: I see a lot of places where ```m_image.Destroy()``` is explicitly called. Is it possible to create a helper method in the glyph class? It will shorten somehow the notation. Also, is it possible to create a helper method in data?